### ISSUE TITLE
request: Postgresql ADD constraint locks table and should give warning

### ISSUE THREAD

**Issue by tned73** [3YOURMIND/django-migration-linter#309]
TLDR; I would like to get a warning on changing/adding a foreign key for Postgresql that it locks the table and if the table is large, it could be locked for some time.

When adding or changing a foreign key field on a table something like the following SQL will be created by the sqlmigrate:

the change is: allowing null on a foreign-key field
```
SET CONSTRAINTS "dummy_customer_id_f949647c_fk_dummy_customer_id" IMMEDIATE; ALTER TABLE "dummy_card" DROP CONSTRAINT "dummy_card_customer_id_f949647c_fk_dummy_customer_id";
ALTER TABLE "dummy_card" ALTER COLUMN "customer_id" DROP NOT NULL;
ALTER TABLE "dummy_card" ADD CONSTRAINT "dummy_card_customer_id_f949647c_fk_dummy_customer_id" FOREIGN KEY ("customer_id") REFERENCES "dummy_customer" ("id") DEFERRABLE INITIALLY DEFERRED;
```

The last ADD CONSTRAINT on postgresql will start checking if all the values in this table are present in the referred table. If these tables are big this could take a significant amount of time to check this. In the meantime the table is locked and normal operation could be halted.

The solution I am using now is rewriting the migration with the SeparateDatabaseAndState and add to the ADD CONSTRAINT command at the end NOT VALID. This addition will prevent Postgresql from validating all the existing values and will only check the new ones.
The values are valid, because we just dropped the constraint, so no need to validate them again.

postgresql documentation: [https://www.postgresql.org/docs/current/sql-altertable.html#SQL-ALTERTABLE-DESC-ADD-TABLE-CONSTRAINT
](https://www.postgresql.org/docs/current/sql-altertable.html#SQL-ALTERTABLE-DESC-ADD-TABLE-CONSTRAINT
)
So technically this is not a problem for backwards compatibility, but a possible locking migration issue.
